src/net.cpp

```diff
@@ -1791,7 +1791,8 @@ void CConnman::CreateNodeFromAcceptedSocket(std::unique_ptr<Sock>&& sock,
     const bool inbound_onion = std::find(m_onion_binds.begin(), m_onion_binds.end(), addr_bind) != m_onion_binds.end();
     // The V2Transport transparently falls back to V1 behavior when an incoming V1 connection is
     // detected, so use it whenever we signal NODE_P2P_V2.
-    const bool use_v2transport(nLocalServices & NODE_P2P_V2);
+    ServiceFlags local_services = GetLocalServices();
```

Comment:
> nit: Seems like this isn't needed so I would have preferred if that was left out or a separate commit because the file isn't touched at all otherwise.

This is needed to prevent a race condition. Since commit fafc73bc036792923ca64a112684abaf78f3b48e, `nLocalServices` can be accessed from different threads. Therefore, we need to store the value once it's read and use it here and few lines below (in the `InitializeNode` call), to maintain consistency in this flow. Otherwise, `use_v2transport` could have been calculated with a different `nLocalServices` value than the one used in `InitializeNode`.